internal/database/memory/writebatch.go

```diff
@@ -0,0 +1,76 @@
+package memory
+
+type operationKind uint8
+
+const (
+	operationSet operationKind = iota
+	operationDelete
+)
+
+type operation struct {
+	kind  operationKind
+	key   string
+	value []byte
+}
+
+// writeBatch implements an in-memory write batch and prefixes
+// all keys with a certain given prefix.
+// It is NOT thread safe, but its Flush operation is thread safe for
+// the database injected.
+type writeBatch struct {
+	prefix     string
+	database   *Database
+	operations []operation
```

Comment:
Note this fixes the current issue in the chaindb batch, where the order of operation is kept.

Reply:
> Are you saying that currently chaindb does not keep order of operations?

Yes 😄 

https://github.com/ChainSafe/chaindb/blob/f900218c88f853b64138c628c2761383f4180dd3/database.go#L89-L90

> Wouldn't that mess up things while executing them?

Yes 😄